### PR TITLE
Adds health check for jooq connections.

### PR DESCRIPTION
**Pull request by alokmenghrajani**

### CODE REVIEW

File: server/src/main/java/keywhiz/ServiceModule.java [hunk 1]
```diff
@@ -84,16 +87,29 @@ public ServiceModule(KeywhizConfig config, Environment environment) {
         bindSecretGenerator("templated", TemplatedSecretGenerator.class);
       }
     });
+
+    // Add Dropwizard health check for readonly database
+    environment.healthChecks().register("postgres-readonly-health", new HealthCheck() {
+      @Override protected Result check() {
```
Comment (sul3n3t) [inline server/src/main/java/keywhiz/ServiceModule.java]: Would be nice to have this be it's own class with a unit test.

Comment (alokmenghrajani) [inline server/src/main/java/keywhiz/ServiceModule.java]: Done

File: server/src/main/java/keywhiz/ServiceModule.java [hunk 2]
```diff
@@ -88,18 +88,15 @@ public ServiceModule(KeywhizConfig config, Environment environment) {
...
-   * the connection is down. Optionally, fallback to read-write if that connection is healthy.
-   */
   @Provides @Singleton ManagedDataSource dataSource(Environment environment,
       KeywhizConfig config) {
     DataSourceFactory dataSourceFactory = config.getDataSourceFactory();
     ManagedDataSource dataSource = dataSourceFactory.build(environment.metrics(), "postgres-writable");
     environment.lifecycle().manage(dataSource);
 
+    environment.healthChecks().register("db-read-write-health",
+        new KeywhizDbHealthCheck(dataSource, true));
```
Comment (sul3n3t) [inline server/src/main/java/keywhiz/ServiceModule.java]: An inline comment like `true /* only log unhealthy */)` would make this more clear. Either that or switching the boolean to an enum.

File: server/src/main/java/keywhiz/KeywhizDbHealthCheck.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+import org.jooq.exception.DataAccessException;
+import org.jooq.impl.DSL;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * TODO: we could improve the code to use the read-write connection if only the readonly connection
+ * is down.
+ */
+public class KeywhizDbHealthCheck extends HealthCheck {
```
Comment (sul3n3t) [inline server/src/main/java/keywhiz/KeywhizDbHealthCheck.java]: How about `JooqHealthCheck`?

File: server/src/main/java/keywhiz/ServiceModule.java (hunk 2 above)
Comment (alokmenghrajani) [inline server/src/main/java/keywhiz/ServiceModule.java]: Switched to enum.

File: server/src/main/java/keywhiz/KeywhizDbHealthCheck.java (hunk 3 above)
Comment (alokmenghrajani) [inline server/src/main/java/keywhiz/KeywhizDbHealthCheck.java]: Renamed.

File: server/src/main/java/keywhiz/ServiceModule.java [hunk 4]
```diff
@@ -50,6 +50,7 @@
 import org.jooq.impl.DSL;
 
 import static com.google.common.base.Preconditions.checkNotNull;
+import static keywhiz.JooqHealthCheck.OnFailure.*;
```
Comment (sul3n3t) [inline server/src/main/java/keywhiz/ServiceModule.java]: No wildcard imports please. You can instruct IntelliJ to enumerate them.

**Comment by sul3n3t** [comment]
LGTM aside from my import comment

File: server/src/main/java/keywhiz/ServiceModule.java (hunk 4 above)
Comment (alokmenghrajani) [inline server/src/main/java/keywhiz/ServiceModule.java]: fixed.

**Comment by sul3n3t** [comment]
LGTM

**Pull request merged**